force-app/main/default/classes/ReviewHandler.cls

```diff
@@ -0,0 +1,111 @@
+public class ReviewHandler {
+    // Prevents reviews for applications with 'Rejected' or 'Pending' status
+    public static void preventReviews(List<Review__c> reviews) {
+        Set<Id> rejectedAppIds = new Set<Id>();
+
+        // Collect the IDs of related applications
+        for (Review__c review : reviews) {
+            if (review.Application__c != null) {
+                rejectedAppIds.add(review.Application__c);
+            }
+        }
+
+        // Query the status of related applications
+        Map<Id, Application__c> appMap = new Map<Id, Application__c>([SELECT Id, Status__c FROM Application__c WHERE Id IN :rejectedAppIds]);
+
+        // Iterate through the reviews to verify the application status
+        for (Review__c review : reviews) {
+            if (review.Application__c != null) {
+                Application__c app = appMap.get(review.Application__c);
+
+                // If the related application status is 'Rejected' or 'Pending,' prevent the review from being created
+                if (app != null && (app.Status__c == 'Rejected' || app.Status__c == 'Pending')) {
+                    review.addError('Cannot create a review for an application with a "Rejected" or "Pending" status.');
+                }
+            }
+        }
+    }
+
+    // Updates positions based on the review status
+    public static void updatePositions(List<Review__c> reviews) {
+        Set<Id> applicationIdsToUpdate = new Set<Id>();
+        Map<Id, Position__c> positionsToUpdate = new Map<Id, Position__c>();
+
+        // Iterate through the reviews to identify records with Status 'Hired'
+        for (Review__c review : reviews) {
+            if (review.Status__c == 'Hired') {
```

Comment:
why are we checking for only current field value?
**for e.g. if i update some field when the status is hired then this will be fired. So please check the prior value and current value so that this condition will run only once.**

Reply:
sir , resolved it. will be pushing all the codes in one go today.

Reply:
done sir